### PR TITLE
Add option to remove memory from sxs waveforms

### PR DESCRIPTION
**Pull request by md-arif-shaikh**
Removing memory may give better results for high mass-ration, and eccentric precessing systems. This PR adds an option for `sxs` waveforms.

### CODE REVIEW

File: gw_eccentricity/load_data.py [hunk 1]
```diff
@@ -526,6 +534,14 @@ def get_defaults_for_nr():
     extrap_order: int
         Extrapolation order to use for loading the waveform data.
         NOTE: This is used only for sxs catalog formatted waveforms.
+
+    remove_memory: bool
```
Comment (vijayvarma392) [inline gw_eccentricity/load_data.py:538]: I wonder if it's best to remove memory by default. So, keep_memory?

File: gw_eccentricity/load_data.py [hunk 2]
```diff
@@ -1093,9 +1127,9 @@ def get_modes_dict_from_sxs_catalog_old_format(**kwargs):
             time = mode_data[:, 0]
             t = np.arange(time[0], time[-1], kwargs["deltaTOverM"])
         hlm = mode_data[:, 1] + 1j * mode_data[:, 2]
-        amp_interp = interpolate(t, time, np.abs(hlm))
-        phase_interp = interpolate(t, time, -np.unwrap(np.angle(hlm)))
-        hlm_interp = amp_interp * np.exp(-1j * phase_interp)
+        real_interp = interpolate(t, time, np.real(hlm))
+        imag_interp = interpolate(t, time, np.imag(hlm))
+        hlm_interp = real_interp + 1j * imag_interp
```
Comment (vijayvarma392) [inline gw_eccentricity/load_data.py:1132]: I forget, what is the motivation for this change?

File: gw_eccentricity/load_data.py [hunk 3]
```diff
@@ -1109,6 +1143,28 @@ def get_modes_dict_from_sxs_catalog_format(**kwargs):
     # get the waveform object
     waveform = sxs.rpdmb.load(
```
Comment (vijayvarma392) [inline gw_eccentricity/load_data.py:1144]: Nowadays, you can just do sxs.load

**Review by vijayvarma392 — CHANGES_REQUESTED** [review]

File: gw_eccentricity/load_data.py (hunk 1 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:538]: Do you suggest `keep_memory` with default set to `True`?

File: gw_eccentricity/load_data.py (hunk 2 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:1132]: In the extreme precession case, intepolating amp and phase on smaller timestep was introducing some artefacts where as real and imaginary was not having those issues.

File: gw_eccentricity/load_data.py (hunk 3 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:1144]: Okay, I will use that.

File: gw_eccentricity/load_data.py (hunk 1 above)
Comment (vijayvarma392) [reply gw_eccentricity/load_data.py:538]: No, with default False, so that by default no memory.

Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py]: Done

File: gw_eccentricity/load_data.py (hunk 2 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:1136]: moved discussion to slack

Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:1139]: Added comments

File: gw_eccentricity/load_data.py [hunk 4]
```diff
@@ -824,6 +843,11 @@ def load_sxs_catalogformat(**kwargs):
         to locate the strain file. This function will seek a file named
         `Strain_N{extrap_order}.h5` in the `data_dir`.
 
+    keep_memory: bool
+        If False, remove memory contribution from the waveform modes.
+        This will require metadata file to find t_relax which is used
+        to start the integration for computing memory contribution.
```
Comment (vijayvarma392) [inline gw_eccentricity/load_data.py:849]: Should it say what the default is?

File: gw_eccentricity/load_data.py [hunk 5]
```diff
@@ -526,6 +535,14 @@ def get_defaults_for_nr():
     extrap_order: int
         Extrapolation order to use for loading the waveform data.
         NOTE: This is used only for sxs catalog formatted waveforms.
+
+    keep_memory: bool
+        If False, remove memory contribution from the waveform modes.  This will
+        require metadata file to find t_relax which is used to start the
+        integration for computing memory contribution.  NOTE: This can be used
+        only in the newer sxs catalog formatted waveforms with
+        origin=`SXSCatalog`
```
Comment (vijayvarma392) [inline gw_eccentricity/load_data.py:544]: Does the code raise a helpful Exception if this is used with catalogs other than SXSCatalog?

**Review by vijayvarma392 — CHANGES_REQUESTED** [review]

File: gw_eccentricity/load_data.py (hunk 5 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:544]: It should by design since it is included only in the list of kwargs for SXSCatalog. I will share an example case.

File: gw_eccentricity/load_data.py (hunk 4 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:849]: Earlier we decided that we are not going to mention the default for each kwarg here and instead refer to `load_data.get_load_waveform_defaults('SXSCatalog')` for getting the defaults. For example at the of of the do string under parameters, we refer to this function. The idea was to set the defaults for NR at a single place because some of these are shared by different waveform origins.

Comment (vijayvarma392) [reply gw_eccentricity/load_data.py:849]: Ok, I'll leave it to you, just make sure it would be easy for a reader to understand, and feel free to veto old decisions if they no longer make sense.

File: gw_eccentricity/load_data.py (hunk 5 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:544]: Here is the error message if `keep_memory` is used, for example, with `SXSCatalog_old`
image

File: gw_eccentricity/load_data.py (hunk 4 above)
Comment (md-arif-shaikh) [reply gw_eccentricity/load_data.py:849]: I don't have a strong opinion about changing things as of now. But we can consider changing the current setup later, possibly in a separate PR.

**Comment by md-arif-shaikh** [comment]
Merge?
